src/components/buttons/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import React from "react";
+import styles from "./radioButton.module.css";
+
+interface RadioButtonProps {
+  id: string;
+  name: string;
+  label: string;
+  value?: string;
+  checked?: boolean;
+  disabled?: boolean;
+  defaultChecked?: boolean; 
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const RadioButton: React.FC<RadioButtonProps> = ({
+  id,
+  name,
+  value,
+  label,
+  checked,
+  disabled,
+  defaultChecked,
+  onChange,
+}) => {
+  return (
+    <div aria-label={label}
+      className={`${styles.main}`}
+    >   
+    <input
+        className={`${styles.input}`}
+        type="radio"
+        name={name}
+        id={id}
+        value={value}
+        checked={checked}
+        disabled={disabled}
```

Comment:
let's look at disabled and aria-disabled, we might actually want to use aria disabled and then add extra js logic to make sure it's function is also disabled. 

with disabled the element is hidden from a11y tech, so a screenreader wouldn't read the element/option. but with aria-diabled it'd read it, but list it as disabled, meaning it's not hidden.

Reply:
I'll look into it!